src/models.ts

```diff
@@ -87,11 +87,22 @@ export class Block extends BaseTransfer<BlockId, BlockDTO> {
 
   $fromDTO = blockFromDTO;
 
+  // TODO this isn't really the create url, its a get?
   $createUrl = () => `/blocks/${this.id.project}/${this.id.block}`;
 
   $recieve = blocks.actions.recieve;
 
   $selector = selectBlock;
+
+  static makeCreateUrl = (projectId: string) => `/blocks/${projectId}`;
+
+  static async create(session: ISession, projectId: string, data: JsonObject): Promise<BlockId> {
+    const { status, json } = await session.post(Block.makeCreateUrl(projectId), data);
+    if (status > 400) throw Error(`Could not create block, status: ${status}`);
+    if (!('id' in json))
+      throw Error(`Could not create block, invalid response: ${JSON.stringify(json)}`);
+    return json.id as BlockId;
+  }
```

Comment:
a static create function is probably the way to go but going to be common across more types.
Move most of this to base?
Also should we return the json package rather than just the id?